pkgs/tools/wayland/xwaylandvideobridge/default.nix

```diff
@@ -0,0 +1,51 @@
+{ lib
+, stdenv
+, fetchFromGitLab
+, cmake
+, extra-cmake-modules
+, pkg-config
+, qt5
+, libsForQt5
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "xwaylandvideobridge";
+  version = "0.2";
+
+  src = fetchFromGitLab {
+    domain = "invent.kde.org";
+    owner = "system";
+    repo = "xwaylandvideobridge";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-79Z4BH7C85ZF+1Zivs3bTey7IdUnaDKhvZxzL6sDqUs=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    extra-cmake-modules
+    pkg-config
+    qt5.wrapQtAppsHook
+  ];
+
+  buildInputs = [
+    qt5.qtbase
+    qt5.qtquickcontrols2
+    qt5.qtx11extras
+    libsForQt5.kdelibs4support
+    libsForQt5.kpipewire
+  ];
+
+  dontWrapQtApps = true;
+
+  preFixup = ''
+    wrapQtApp "$out/bin/xwaylandvideobridge"
+  '';
```

Comment:
This doesn't make much sense.
 Is there something I'm missing or could you remove both of these?

Reply:
The package will build without these two lines, but [the following error](https://nixos.wiki/wiki/Qt#:~:text=%3B%0A%20%20%20%20%7D%3B%0A%20%20%7D%3B-,qt.qpa.plugin%3A%20Could%20not%20find%20the%20Qt%20platform%20plugin%20%22xcb%22%20in%20%22%22,-Here%20is%20a) will occur:

```shell
$ ./result/bin/xwaylandvideobridge
qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
```

Reply:
@stepbrobd can you try this 
```nix
{ lib
, mkDerivation
, fetchFromGitLab
, cmake
, extra-cmake-modules
, pkg-config
, qtbase
, qtquickcontrols2
, qtx11extras
, kdelibs4support
, kpipewire
, wrapQtAppsHook
}:
let 
  version = "0.2";
in
mkDerivation {
  pname = "xwaylandvideobridge";
  inherit version;

  src = fetchFromGitLab {
    domain = "invent.kde.org";
    owner = "system";
    repo = "xwaylandvideobridge";
    rev = "v${version}";
    hash = "sha256-79Z4BH7C85ZF+1Zivs3bTey7IdUnaDKhvZxzL6sDqUs=";
  };

  nativeBuildInputs = [
    cmake
    extra-cmake-modules
    pkg-config
    wrapQtAppsHook
  ];

  buildInputs = [
    qtbase
    qtquickcontrols2
    qtx11extras
    kdelibs4support
    kpipewire
  ];

  meta = {
    description = "Utility to allow streaming Wayland windows to X applications";
    homepage = "https://invent.kde.org/system/xwaylandvideobridge";
    license = lib.licenses.gpl2Plus;
    maintainers = with lib.maintainers; [ stepbrobd ];
    platforms = lib.platforms.linux;
    mainProgram = "xwaylandvideobridge";
  };
}
```

Reply:
oh and i was mistaken it's not `qt5.callPackage` it's `libsForQt5.callPackage`

Reply:
Yours worked without
```nix
  dontWrapQtApps = true;

  preFixup = ''
    wrapQtApp "$out/bin/xwaylandvideobridge"
  '';
  ```

Then I went back and removed these lines from my version, it also worked. I'm curious what caused this inconsistency 😂. I'll add `mainProgram = "xwaylandvideobridge";`, then figure out if xwaylandvideobridge can be built with [Qt6](https://nixos.wiki/wiki/Qt#:~:text=when%20it%27s%20required.%0A%20%20%27%27%3B%0A%7D-,Migrating%20apps%20from%20Qt5%20to%20Qt6,-Replace%20libsForQt5.callPackage).

Reply:
About `finalAttrs` pattern: https://github.com/NixOS/nixpkgs/pull/240625#discussion_r1247474967

Reply:
I know about `finalAttrs`, it doesn't work with `libsForQt6.mkDerivation` which is what allowed the removal of the wrapping bits

Reply:
Thanks for the explanation!

Should we use `libsForQt5.libsForQt5` with `stdenv.mkDerivation` + `finalAttrs` pattern here, or `libsForQt5.mkDerivation`?

The wrapping bits can be removed with either way.

Reply:
For future reference @stepbrobd, please build and test packages locally before sending a PR. This can be done using `nix-build -A xwaylandvideobridge` in the top of your local nixpkgs clone.

Reply:
In this case you can use libsForQt5.callPackage with stdenv.mkDerivation + finalAttrs

Reply:
Thanks. Also, the initial version was built and tested locally before sending this PR, the conversation here is just a part of the review process.